common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsAuthorizationResultFactory.java

```diff
@@ -0,0 +1,95 @@
+package com.microsoft.identity.common.internal.providers.microsoft.microsoftsts;
+
+import android.content.Intent;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.adal.internal.util.StringExtensions;
+import com.microsoft.identity.common.exception.ErrorStrings;
+import com.microsoft.identity.common.internal.logging.Logger;
+import com.microsoft.identity.common.internal.providers.microsoft.MicrosoftAuthorizationErrorResponse;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationErrorResponse;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResult;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResultFactory;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationStatus;
+import com.microsoft.identity.common.internal.util.StringUtil;
+
+import java.util.HashMap;
+
+public class MicrosoftStsAuthorizationResultFactory extends AuthorizationResultFactory {
```

Comment:
See [this comment](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/133/files/34372525a1342e7fc0e5ff69a49a0e871423275a#r195851360) and consider the following class declaration

```java
public class MicrosoftStsAuthorizationResultFactory extends AuthorizationResultFactory<MicrosoftStsAuthorizationResult> {
    // ...
}
```